src/components/TextMaskedInput/TextMaskedInput.js

```diff
@@ -129,9 +135,21 @@ export const TextMaskedInput = (props) => {
         allCaps={allCaps}
         capitalize={capitalize}
       />
-
-      <MaskedInput {...maskedInputProps} />
-
+      <div className={styles.TextInputWrapper}>
+        {iconPrefix && iconName ? (
```

Comment:
@MayUWish same concern as above. The validation here should be more specific.

Reply:
Same as above.  invalid input => no icon.
<img width="1035" alt="image" src="https://user-images.githubusercontent.com/84494313/163069863-878d6676-b0ef-40bf-8766-06f5bfc75121.png">